src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/LocationsControllerTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using SFA.DAS.ApprenticeAan.Api.Controllers;
+using SFA.DAS.ApprenticeAan.Application.Locations.Queries.GetAddresses;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Api.UnitTests.Controllers;
+
+public class LocationsControllerTests
+{
+    [Test]
+    [MoqAutoData]
+    public async Task GetAddresses_ReturnsAddresses(
+        GetAddressesQueryResult response,
+        [Frozen] Mock<IMediator> mockMediator,
+        [Greedy] LocationsController sut)
+    {
+        mockMediator
+            .Setup(m => m.Send(It.IsAny<GetAddressesQuery>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(response);
+
+        var result = await sut.GetAddresses("thisIsAQuery");
+
+        result.As<OkObjectResult>().Should().NotBeNull();
+
+        result.As<OkObjectResult>().Value.Should().BeEquivalentTo(response);
+    }
+
+    [Test]
+    [MoqAutoData]
+    public async Task GetAddresses_NoMatch_ReturnNotFoundResponse(
+        [Frozen] Mock<IMediator> mockMediator,
+        [Greedy] LocationsController sut)
+    {
+        var response = new GetAddressesQueryResult();
```

Comment:
Minor: Creation of response object can be simplified using MoqAutoData used in previous test case **Task GetAddresses_ReturnsAddresses**

Reply:
I didn't want the property in response to be auto populated, as I am testing empty response.